backend/routes/Adminusers.js

```diff
@@ -0,0 +1,114 @@
+var express = require('express');
+var router = express.Router();
+const bodyParser = require('body-parser');
+var User = require('../models/Adminuser');
+var passport = require('passport');
+var authenticate = require('../Adminauthenticate');
+const { TokenExpiredError } = require('jsonwebtoken');
+const cors = require('./cors');
+
+router.use(bodyParser.json());
+
+/* GET users listing. */
+router.options('*', cors.corsWithOptions, (req, res) => { res.sendStatus(200); });
+router.get('/', cors.corsWithOptions, authenticate.verifyUser, authenticate.verifyAdmin, (req, res, next) => {
+  User.find({}, (err, users) => {
+    if (err) {
+      return next(err);
+    } else {
+      res.statusCode = 200;
+      res.setHeader('Content_type', 'application/json');
+      res.json(users);
+    }
+  })
+});
+
+router.post('/signup', cors.corsWithOptions, (req, res, next) => {
+  User.register(new User({ username: req.body.username }),
+    req.body.password, (err, user) => {
+      if (err) {
+        res.statusCode = 500;
+        res.setHeader('Content-Type', 'application/json');
+        res.json({ err: err });
+      }
+      else {
+        if(req.body.admin){
+          user.admin = req.body.admin
+        }
+        if(req.body.email){
+          user.email = req.body.email
+        }
+        user.save((err, user) => {
+          if (err) {
+            res.statusCode = 500;
+            res.setHeader('Content-Type', 'application/json');
+            res.json({ err: err });
+            return;
+          }
+          passport.authenticate('local')(req, res, () => {
+            res.statusCode = 200;
+            res.setHeader('Content-Type', 'application/json');
+            res.json({ success: true, status: 'Registration Successful!' });
+          });
+        });
+      }
+    });
+});
+
+router.post('/login', cors.corsWithOptions, (req, res, next) => {
```

Comment:
Hey @Aaryan8751 I am able to login even if I am not an admin. Please apply the check that if `admin` field is `false` don't allow the user to perform any action except signup